dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpTransport.java

```diff
@@ -191,7 +192,7 @@ public void sendError(Exception e) throws IOException {
      * @param e The JSONRPCException to be included in the error response.
      */
     public void sendJsonRpcException(JSONRPCException e) {
-        McpResponse mcpResponse = new McpErrorResponse(mcpRequest == null ? "" : mcpRequest.id(), e);
+        McpResponse mcpResponse = new McpErrorResponse(mcpRequest == null ? new McpRequestId("") : mcpRequest.id(), e);
         jsonb.toJson(mcpResponse, writer);
     }
```

Comment:
I didn't notice this before, but if the `id` could not be parsed from the request, [the response must set the `id` to JSON `null`](https://www.jsonrpc.org/specification#response_object).

Maybe we could have a special instance of `McpRequestId` which has `value` set to `JsonValue.NULL`?

Reply:
I've resolved this by creating an `McpRequestId` with an empty string `""`. Then in the serialiser, if the ID doesn't have any values or is empty, we return null. Then for anything that serialises the `McpRequestId`, I used `@JsonbNillable` so the JSON response could return `"id":null`

Reply:
I think either of those options would be sufficient, you don't need both.